Mappers/HttpMapper.php

```diff
@@ -191,36 +228,91 @@ protected function getHttpMapperConfig($urlName)
     }
 
     /**
-     * @param array $httpMapperConfig
-     * @return HttpMapper
+     * @param int $statusCode
+     * @throws HttpResourceNotFoundException
+     * @throws InfrastructureException
+     * @throws ValidationException
      */
-    private function setHttpMapperConfig($httpMapperConfig)
+    private function throwException(int $statusCode)
     {
-        $this->httpMapperConfig = $httpMapperConfig;
+        switch (true) {
+            case $statusCode == 404:
+                throw new HttpResourceNotFoundException('Http resource not found');
+
+                break;
+            case $statusCode >= 400 && $statusCode < 500:
+                throw new ValidationException([]);
```

Comment:
What does it mean - validation exception?

Reply:
What for you need that function?